politeiawww/util/proposal_test.go

```diff
@@ -0,0 +1,148 @@
+package util_test
+
+import (
+	"testing"
+
+	www "github.com/decred/politeia/politeiawww/api/v1"
+	"github.com/decred/politeia/politeiawww/util"
+)
+
+func TestProposalName(t *testing.T) {
+	var (
+		expectedTitle = "this-is-the-title"
```

Comment:
We should not mangle the title.  We had debates on slack to not use - for spaces.  So please make the title dashless and let's run it again through test.

Reply:
hmm, in that case we need to write new logic - maybe something inspired in https://github.com/writeas/go-strip-markdown with our own regex rules.

Reply:
It doesn't need to be complicated.  Let me give you a few examples:
###This is a title -> This is a title
   This-is a-Title -> This-is a-Title
\n\n#This-is MY  tittle### -> This-is MY  tittle

Reply:
for future reference:
ProposalName func: name is the first line
\n\n#This-is MY tittle### -> ""
Strip func:
\n\n#This-is MY tittle### -> "This-is MY tittle"